tests/ui/lint/lint-ctypes-enum.stderr

```diff
@@ -45,21 +45,131 @@ note: the type is defined here
 LL | enum T {
    | ^^^^^^
 
-error: `extern` block uses type `u128`, which is not FFI-safe
+error: `extern` block uses type `Option<NonZero<u8>>`, which is not FFI-safe
```

Comment:
No idea where this is coming from, whether this is just a diagnostics bug, or how to fix it.

Reply:
I did a code review of the lint in https://github.com/rust-lang/rust/blob/4cdd20584ccb75890d7d9bfae266054abfae5d46/compiler/rustc_lint/src/types.rs#L987 but it looks alright. So to debug it we'll need to add some tracing and figure out where it goes wrong